modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,388 @@
+  ) {
+    val minFirstOrdinalValue = 0L
+    val maxFirstOrdinalValue = ordinalTrustUpdateInterval - 1L
+    val minSecondOrdinalValue = ordinalTrustUpdateInterval.value + ordinalTrustUpdateDelay
+    val maxSecondOrdinalValue = (ordinalTrustUpdateInterval * 2) - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+
+      firstDelta <- Gen.chooseNum(minFirstOrdinalValue, maxFirstOrdinalValue)
+      firstOrdinal = Refined.unsafeApply[Long, NonNegative](firstDelta)
+
+      secondDelta <- Gen.chooseNum(minSecondOrdinalValue, maxSecondOrdinalValue)
+      secondOrdinal = Refined.unsafeApply[Long, NonNegative](secondDelta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(firstOrdinal), SnapshotOrdinal(secondOrdinal))
+
+    forall(gen) {
+      case (trust, firstOrdinal, secondOrdinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          _ <- store.updateNext(firstOrdinal)
+
+          firstNext <- store.getNextOrdinalTrust
+
+          _ <- store.updateCurrent(secondOrdinal)
+
+          current <- store.getCurrentOrdinalTrust
+          secondNext <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            Order.lteqv(SnapshotOrdinal(ordinalTrustUpdateInterval |+| ordinalTrustUpdateDelay), secondOrdinal),
+            firstNext.trust === current.trust,
+            SnapshotOrdinal(ordinalTrustUpdateInterval) === current.ordinal,
+            PublicTrustMap.empty === current.peerLabels,
+            SnapshotOrdinal(Refined.unsafeApply[Long, NonNegative](ordinalTrustUpdateInterval * 2)) === secondNext.ordinal,
+            isEmpty(secondNext.trust)
+          )
+    }
+  }
+
+  test(
+    "next ordinal trust storage doesn't update from gossip when the gossip ordinal doesn't match the next ordinal"
+  ) {
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      peerId <- peerIdGen
+    } yield (TrustMap.empty.copy(trust = trustInfo), peerId)
+
+    forall(gen) {
+      case (trust, peerId) =>
+        for {
+          store <- TrustStorage.make(trust)
+          gossipOrdinal = SnapshotOrdinal(2000L)
+
+          firstNext <- store.getNextOrdinalTrust
+          gossipOrdinalPublicTrust = SnapshotOrdinalPublicTrust(
+            gossipOrdinal,
+            trust.toPublicTrust
+          )
+
+          _ <- store.updateNext(peerId, gossipOrdinalPublicTrust)
+
+          secondNext <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            firstNext.ordinal =!= gossipOrdinal,
+            isEmpty(firstNext.trust),
+            firstNext.peerLabels === PublicTrustMap.empty,
+            firstNext.ordinal === secondNext.ordinal,
+            firstNext.trust === secondNext.trust,
+            firstNext.peerLabels === secondNext.peerLabels
+          )
+    }
+  }
+
+  test(
+    "next ordinal trust storage doesn't update from gossip because an entry already exist for the peer id"
```

Comment:
as above

Reply:
Changed to `next ordinal trust storage doesn't update from gossip when an entry already exists for the peer id`.